src/methods/wallet-send-calls.ts

```diff
@@ -0,0 +1,80 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { Infer } from '@metamask/superstruct';
+import {
+  nonempty,
+  type,
+  string,
+  array,
+  object,
+  optional,
+  tuple,
+} from '@metamask/superstruct';
+import type {
+  Json,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from '@metamask/utils';
+import { HexChecksumAddressStruct, StrictHexStruct } from '@metamask/utils';
+
+import {
+  validateAndNormalizeKeyholder,
+  validateParams,
+} from '../utils/validation';
+
+const SendCallsStruct = tuple([
+  object({
+    version: optional(nonempty(string())),
+    from: HexChecksumAddressStruct,
+    chainId: optional(StrictHexStruct),
+    calls: array(
+      object({
+        to: optional(HexChecksumAddressStruct),
+        data: optional(StrictHexStruct),
+        value: optional(StrictHexStruct),
+      }),
+    ),
+    capabilities: optional(type({})),
+  }),
+]);
+
+export type SendCallsParams = Infer<typeof SendCallsStruct>;
+export type SendCalls = SendCallsParams[0];
+
+export type ProcessSendCallsHook = (
+  sendCalls: SendCalls,
+  req: JsonRpcRequest,
+) => Promise<string>;
+
+export async function walletSendCalls(
+  req: JsonRpcRequest,
+  res: PendingJsonRpcResponse<Json>,
+  {
+    getAccounts,
+    processSendCalls,
+  }: {
+    getAccounts: (req: JsonRpcRequest) => Promise<string[]>;
+    processSendCalls?: ProcessSendCallsHook;
+  },
+): Promise<void> {
+  if (!processSendCalls) {
```

Comment:
should this throw is getAccounts is not provided as well?

Reply:
This is validated when `createWalletMiddleware` is called.